### PR TITLE
DMP-1636: Implement POST /security-groups

### PR DESCRIPTION
**Pull request by chrisbellingham-hmcts**
# [DMP-1636](https://tools.hmcts.net/jira/browse/DMP-1636)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/usermanagement/service/impl/SecurityGroupServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+import java.util.Collections;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class SecurityGroupServiceImpl implements SecurityGroupService {
+
+    private final SecurityGroupRepository securityGroupRepository;
+    private final SecurityRoleRepository roleRepository;
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/usermanagement/service/impl/SecurityGroupServiceImpl.java:23]: Should we call this variable `securityRoleRepository` in keep with the one above?

File: src/main/java/uk/gov/hmcts/darts/usermanagement/mapper/impl/SecurityGroupMapper.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    unmappedSourcePolicy = ReportingPolicy.IGNORE,
+    unmappedTargetPolicy = ReportingPolicy.IGNORE)
+public interface SecurityGroupMapper {
+
+    @Mappings({
+        @Mapping(source = "name", target = "groupName"),
+        @Mapping(source = "displayName", target = "groupDisplayName"),
+        @Mapping(source = "description", target = "description")
+    })
+    SecurityGroupEntity mapToUserEntity(SecurityGroup user);
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/usermanagement/mapper/impl/SecurityGroupMapper.java:21]: Should this be called `mapToSecurityGroupEntity`?

Comment (richardbull) [reply src/main/java/uk/gov/hmcts/darts/usermanagement/mapper/impl/SecurityGroupMapper.java:21]: also need to rename the method param - user

File: src/main/resources/openapi/usermanagement.yaml [hunk 3]
```diff
@@ -379,7 +379,7 @@ components:
         id:
           $ref: '#/components/schemas/SecurityGroupId'
         role:
```
[the remark this answers is not in the document]
Comment (chrisbellingham-hmcts) [reply src/main/resources/openapi/usermanagement.yaml]: Renamed to the more descriptive `role_id`

**Review by richardbull — APPROVED** [review]

**Review by tonymort — APPROVED** [review]

**Pull request merged**